data/syntax-highlighting/vim/ftplugin/meson.vim

```diff
@@ -12,9 +12,30 @@ set cpo&vim
 
 setlocal commentstring=#\ %s
 setlocal comments=:#
+setlocal formatoptions+=croql formatoptions-=t
 
-setlocal shiftwidth=2
-setlocal softtabstop=2
+let b:undo_ftplugin = "setl com< cms< fo<"
+
+if get(g:, "meson_recommended_style", 1)
+  setlocal expandtab
+  setlocal shiftwidth=2
+  setlocal softtabstop=2
+  let b:undo_ftplugin .= " | setl et< sts< sw<"
```

Comment:
In theory this could probably append to the previous setl, rather than starting a new one... would require actually making sure undo_ settings are defined in the right order, not sure it is worth it.

Reply:
Yes, it becomes more fragile in larger and more complicated plugins.  I updated over a hundred so I was trying to follow a consistent strategy.

Reply:
That's a reasonable approach. :)